Makefile

```diff
@@ -58,8 +58,13 @@ checksum:
 clean:
 	@rm -rf $(BINDIR) ./_dist
 
+GIT_BASEDIR    = $(shell git rev-parse --show-toplevel 2>/dev/null)
+ifneq ($(GIT_BASEDIR),)
+	LDFLAGS += -X github.com/Azure/acs-engine/pkg/test.JUnitOutDir=${GIT_BASEDIR}/test/junit
+endif
+
 test: test-style
-	go test -v $(GOFILES)
+	go test -v -ldflags '$(LDFLAGS)' $(GOFILES)
```

Comment:
Or since it seems ginkgo runs all go tests too, how about switching the `make test` to run `ginkgo -v -ldflags='$(LDFLAGS)' $(GOFILES)` only?  Should we add any switches defined here: https://onsi.github.io/ginkgo/#ginkgo-and-continuous-integration